dropshot_endpoint/src/doc.rs

```diff
@@ -227,13 +227,21 @@ mod tests {
         ///
         /// Even
         /// More
+        ///
+        ///
+        ///
+        /// And another
+        /// paragraph
         #[derive(Schema)]
         struct SummaryDescriptionBreak;
         assert_eq!(
             ExtractedDoc::from_attrs(&SummaryDescriptionBreak::schema().attrs),
             ExtractedDoc {
                 summary: Some("Summary".to_string()),
-                description: Some("Text More\nEven More".to_string())
+                description: Some(
+                    "Text More\n\nEven More\n\nAnd another paragraph"
+                        .to_string()
+                )
```

Comment:
Huh, I don't quite understand -- were other doc comments not similarly impacted by this? Or have we been working around this for a while?

Also, does Rust collapse down multiple blank lines? Naively I'd have expected six `\n` instances here.

Reply:
At least in the external API, until I added one yesterday, we had zero endpoints that included a paragraph break other than the one between the first line and the rest.

Regarding the collapsing, I did not know why it was doing that, so I'm glad you asked and made me figure it out. The reason it's skipping subsequent blank lines is that we're hitting the `ends_with('\n')` on line 50, so we're just concatenating `line`, which is `""` in the case of a blank line, so it's a noop.

https://github.com/oxidecomputer/dropshot/blob/7b594d01f47ca783c5d4a25ca2b256602580fe92/dropshot_endpoint/src/doc.rs#L46-L65

While this is correct in outcome, it is very hard to tell that from looking at it, so I'll try tweaking the code.

Reply:
Comment in b868ba22b85e6fca2b51c8d589b3c5d4712482ab is the best I could do :|

Reply:
Gotcha -- so we just look at `acc` and do the collapsing there.